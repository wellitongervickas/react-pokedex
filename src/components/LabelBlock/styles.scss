@import '../../assets/scss/variables';

.label-block {
  h4 {
    margin-bottom: $size-small4;
    text-transform: capitalize;
  }

  margin-bottom: $size-default;

  &:last-of-type {
    margin-bottom: 0;
  }
}
