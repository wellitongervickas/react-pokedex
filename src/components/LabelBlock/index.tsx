import React, { ReactNode } from 'react';

import 'components/LabelBlock/styles.scss';

interface LabelBlockInterface {
  children: ReactNode,
  title?: string,
};

const LabelBlock = ({ title, children }: LabelBlockInterface) => (
  <div className="label-block">
    {title && <h4>{title}</h4>}
    <div className="label-block-content">
      {children}
    </div>
  </div>
);

export default LabelBlock;
