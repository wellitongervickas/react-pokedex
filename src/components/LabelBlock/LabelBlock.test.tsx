import React from 'react';
import { render } from '@testing-library/react';

import LabelBlock from '.';

describe('Component LabelBlock', () => {
  it('renders LabelBlock component', () => {
    const { container, getByText } = render(<LabelBlock title="Hi">Hello World</LabelBlock>);

    const tree = container.firstElementChild;

    expect(getByText('Hi')).toBeInTheDocument();
    expect(getByText('Hello World')).toBeInTheDocument();

    expect(tree?.classList.contains('label-block')).toBeTruthy();
    expect(tree).toMatchSnapshot();
  });

  it('should render LabelBlock without title', () => {
    const { container } = render(<LabelBlock>Hello World</LabelBlock>);

    const tree = container?.firstElementChild?.children[0];
    expect(tree?.classList.contains('label-block-content')).toBeTruthy();

    expect(tree).toMatchSnapshot();
  });
});
