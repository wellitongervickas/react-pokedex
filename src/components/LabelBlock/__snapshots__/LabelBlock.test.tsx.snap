// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component LabelBlock renders LabelBlock component 1`] = `
<div
  class="label-block"
>
  <h4>
    Hi
  </h4>
  <div
    class="label-block-content"
  >
    Hello World
  </div>
</div>
`;

exports[`Component LabelBlock should render LabelBlock without title 1`] = `
<div
  class="label-block-content"
>
  Hello World
</div>
`;
